packages/opentelemetry-instrumentation-http/src/types.ts

```diff
@@ -67,6 +71,14 @@ export interface HttpResponseCustomAttributeFunction {
   (span: Span, response: IncomingMessage | ServerResponse): void;
 }
 
+export interface StartIncomingSpanCustomAttributeFunction {
+  (request: IncomingMessage ): SpanAttributes;
+}
+
+export interface StartOutgoingSpanCustomAttributeFunction {
+  (request: RequestOptions ): SpanAttributes;
+}
+
```

Comment:
In the specs meeting it looks like there will be a different way to doing this (if I'm reading and listening correctly)

https://github.com/open-telemetry/oteps/blob/8335c9a53cdf3398075dd9523d3752522c0ece28/text/0165-instrumentation-api.md

Reply:
I believe i read it the same as you, this is welcome as a lot of instrumentation had some kind of system like this already. I believe this should be handled in the base instrumentation class though. I think it make sense to create a dedicated issue for this

Reply:
👍 
Before it's handled in the base instrumentation, will these startHooks for http be accepted? For I wanna use some custom fields to make the  sample decision in my project. :)

Reply:
I think so